### PR TITLE
Add nextstrain-automation build-configs to phylogenetic worklfow

### PR DESCRIPTION
**Pull request by j23414**
## Description of proposed changes

Adds a custom config and rules for deploying the final Auspice dataset of the phylogenetic workflow.

From [slack](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1711499690732839?thread_ts=1711066498.238259&cid=C01LCTT7JNN), had permission to duplicate changes from https://github.com/nextstrain/zika/pull/51

## Related issue(s)

* Tried to exactly duplicate: https://github.com/nextstrain/zika/pull/51

## Checklist

- [ ] Checks pass

### CODE REVIEW

**Review by joverlee521 — APPROVED** [review]
Thanks for following up here, changes look good by inspection!

File: phylogenetic/build-configs/nextstrain-automation/deploy.smk [hunk 1]
```diff
@@ -0,0 +1,15 @@
+"""
+This part of the workflow handles automatic deployments of the zika build.
```
Comment (victorlin) [inline phylogenetic/build-configs/nextstrain-automation/deploy.smk:2]:
```suggestion
This part of the workflow handles automatic deployments of the dengue build.
```

Comment (j23414) [reply phylogenetic/build-configs/nextstrain-automation/deploy.smk:2]: @victorlin @joverlee521 , could we generalize this and just say "pathogen" build?

This way there's less to trip-over when applying these rules to other pathogens of interest.

Comment (j23414) [reply phylogenetic/build-configs/nextstrain-automation/deploy.smk:2]: Thanks, submitted as a hotfix: https://github.com/nextstrain/dengue/commit/b2c36557d6c6b209d512dec7da10a4f0d8cc7b70

Comment (victorlin) [reply phylogenetic/build-configs/nextstrain-automation/deploy.smk:2]: Good idea, +1 for less pathogen-specific stuff.

**Pull request merged**